sklearn/manifold/_isomap.py

```diff
@@ -159,6 +161,51 @@ class Isomap(_ClassNamePrefixFeaturesOutMixin, TransformerMixin, BaseEstimator):
     (100, 2)
     """
 
+    _parameter_constraints = {
+        "n_neighbors": [Interval(Integral, 0, None, closed="left"), None],
+        "radius": [Interval(Real, 0.0, None, closed="left"), None],
+        "n_components": [Interval(Integral, 1, None, closed="left"), None],
+        "eigen_solver": [StrOptions({"auto", "dense", "arpack"})],
+        "tol": [Interval(Real, 0.0, None, closed="left")],
+        "max_iter": [Interval(Integral, 0, None, closed="left"), None],
+        "path_method": [StrOptions({"auto", "FW", "D"})],
+        "neighbors_algorithm": [StrOptions({"auto", "brute", "kd_tree", "ball_tree"})],
+        "n_jobs": [Integral, None],
+        "metric": [
+            StrOptions(
+                {
+                    "minkowski",
+                    "cityblock",
+                    "cosine",
+                    "euclidean",
+                    "l1",
+                    "l2",
+                    "manhattan",
+                    "braycurtis",
+                    "canberra",
+                    "chebyshev",
+                    "correlation",
+                    "dice",
+                    "hamming",
+                    "jaccard",
+                    "mahalanobis",
+                    "minkowski",
+                    "rogerstanimoto",
+                    "russellrao",
+                    "seuclidean",
+                    "sokalmichener",
+                    "sokalsneath",
+                    "sqeuclidean",
+                    "yule",
```

Comment:
I believe you can get these with `get_scorer_names()`, which can be found in `metrics`.

Reply:
Very helpful, thank you.